src/libNode/Node.cpp

```diff
@@ -611,6 +611,15 @@ bool Node::StartRetrieveHistory(const SyncType syncType,
     return false;
   }
 
+  if (LOOKUP_NODE_MODE && ARCHIVAL_LOOKUP &&
```

Comment:
should it be `or` instead of `and` for `LOOKUP_NODE_MODE && ARCHIVAL_LOOKUP`

Reply:
`LOOKUP_NODE_MODE && ARCHIVAL_LOOKUP` means `newlookup` and `level2lookup` types.
`LOOKUP_NODE_MODE && !ARCHIVAL_LOOKUP` means `lookup` types